GitUI/CommandsDialogs/SettingsDialog/Pages/ScriptsSettingsPage.ScriptInfoProxy.cs

```diff
@@ -76,10 +84,30 @@ private class ScriptInfoProxy
 
             internal void SetImages(ImageList images)
             {
+                if (IconPathName is not null)
+                {
+                    ScriptInfo scriptInfo = this;
+                    Bitmap? icon = scriptInfo.GetIcon();
+                    if (icon is not null && !images.Images.ContainsKey(IconPathName))
+                    {
+                        images.Images.Add(IconPathName, icon.AdaptLightness());
+                    }
+                }
+
                 ImageList = images;
             }
 
-            [return: NotNullIfNotNull("script")]
+            internal string? GetIconImageKey()
+            {
+                if (File.Exists(IconPathName))
+                {
+                    return IconPathName;
+                }
+
+                return Icon;
+            }
```

Comment:
Why is it marked as internal? Feels like we're leaking implementation details.

Why is it a method? This should be a property.
Why not call it `ImageKey`?

Reply:
`internal` was an accident. As for why this isn't a method is because it touches the disk and a rule of thumb that I am trying to follow is that only light CPU workloads are allowed in properties. If you can notice activity using `procmon` then it's a method :) I can still change this into a property if you insist though.